libpromises/processes_select.c

```diff
@@ -1248,6 +1272,187 @@ const char *GetProcessTableLegend(void)
     }
 }
 
+#ifdef __sun
+static FILE *OpenUcbPsPipe(void)
+{
+    struct stat statbuf;
+    if (stat(UCB_PS, &statbuf) < 0)
+    {
+        Log(LOG_LEVEL_VERBOSE,
+            "%s not found, extra process information not available",
+            UCB_PS);
+        return NULL;
+    }
+    if (!(statbuf.st_mode & 0111))
+    {
+        Log(LOG_LEVEL_VERBOSE,
+            "%s not executable, extra process information not available",
+            UCB_PS);
+        return NULL;
+    }
+
+    FILE *cmd = cf_popen(UCB_PS " " UCB_PS_ARGS, "rt", false);
+    if (!cmd)
+    {
+        Log(LOG_LEVEL_WARNING, "Could not execute \"%s %s\", extra process "
```

Comment:
> The message should mention the most serious side-effect: process command line matching will be limited to 80 characters.

Yes, good idea, will add that.

> Additionally, maybe WARNING is too much, given that Solari are System V UNIX and are not guaranteed to have BSD tools.

Note that this message is only if the tool exists, but the execution fails for some reason. So I think the much more common case of not having the tool is covered further up by the `stat` test.

> Last, I've heard recently that `/usr/ucb/ps` will be deprecated on some latest solaris, but the system `ps` works correctly now so it's not needed. We could ping the Illumos experts from the community for more info, do you remember their nicknames?

I lost the link now, but I read a blog post from one of the Solaris developers that `/bin/ps` would be able to do everything `/usr/ucb/ps` can. However, this is *very* recent, and only happened in February/March or so. So many systems will not have it for a while, most likely.

As long as the UCB style ps calls do not report *wrong* information, it does no harm to call it, just a small performance loss, but it will end up reporting exactly the same things.